rosbag2_transport/test/rosbag2_transport/mock_keyboard_handler.hpp

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_TRANSPORT__MOCK_KEYBOARD_HANDLER_HPP_
+#define ROSBAG2_TRANSPORT__MOCK_KEYBOARD_HANDLER_HPP_
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "keyboard_handler/keyboard_handler.hpp"
+
+// press one specific key code one second at a time
+class MockKeyboardHandlerImpl : public KeyboardHandler
```

Comment:
Nit: `Impl` is usually reserved for "hidden" implementations that are only used by a public interface - such as in the pimpl pattern. In this case, I would just call this `MockKeyboardHandler`

```suggestion
class MockKeyboardHandler : public KeyboardHandler
```